homeassistant/components/zwave_js/api.py

```diff
@@ -778,12 +789,30 @@ def device_registered(device: dr.DeviceEntry) -> None:
     msg[DATA_UNSUBSCRIBE] = unsubs
 
     try:
-        result = await controller.async_begin_inclusion(
-            INCLUSION_STRATEGY_NOT_SMART_START[inclusion_strategy.value],
-            force_security=force_security,
-            provisioning=provisioning,
-            dsk=dsk,
-        )
+        if controller.inclusion_state == InclusionState.INCLUDING:
+            connection.send_result(
+                msg[ID],
+                True,  # Inclusion is already in progress
+            )
+            # Check for nodes that have been added but not fully included
+            for node in controller.nodes.values():
+                if node.status != NodeStatus.DEAD and not node.ready:
+                    forward_node_added(
+                        node,
+                        not node.is_secure,
+                        None,
+                    )
+        else:
+            result = await controller.async_begin_inclusion(
+                INCLUSION_STRATEGY_NOT_SMART_START[inclusion_strategy.value],
+                force_security=force_security,
+                provisioning=provisioning,
+                dsk=dsk,
+            )
+            connection.send_result(
+                msg[ID],
+                result,
+            )
     except ValueError as err:
```

Comment:
Please minimize the lines that are included in this try... except block. It was only `controller.async_begin_inclusion` that needed it before. I don't think that has changed.